src/object/interrupt.c

```diff
@@ -263,6 +268,14 @@ void handleInterrupt(irq_t irq)
      * that for any interrupt reported by the platform specific code the generic
      * kernel code does report here that it is done with handling it. */
     ackInterrupt(irq);
+#ifdef CONFIG_ARM_GIC_V3_SUPPORT
+    /* GICv3 uses split priority drop from deactivation, so need to explicitly
+     * deactivate non-forwarded interrupts after they're acknowledged.
+     */
+    if (intStateIRQTable[IRQT_TO_IDX(irq)] != IRQSignal) {
+        deactivateInterrupt(irq);
```

Comment:
There probably needs to be an isb() before `ICC_DIR_EL1` to synchronize with `ICC_EOIR0_EL1`

Reply:
This is what the binary looks like for the relevant part of handleInterrupt:
```
ffffff804001544c:       d518cc33        msr     icc_eoir1_el1, x19
ffffff8040015450:       91124294        add     x20, x20, #0x490
ffffff8040015454:       f0000000        adrp    x0, ffffff8040018000 <ksDomSchedule+0x478>
ffffff8040015458:       d2807fe1        mov     x1, #0x3ff                      // #1023
ffffff804001545c:       f9050801        str     x1, [x0, #2576]
ffffff8040015460:       f8737a80        ldr     x0, [x20, x19, lsl #3]
ffffff8040015464:       f100041f        cmp     x0, #0x1
ffffff8040015468:       54000040        b.eq    ffffff8040015470 <handleInterrupt+0xa0>  // b.none
ffffff804001546c:       d518cb33        msr     icc_dir_el1, x19
```

Reply:
Agreed, but only for the call in handleInterrupt, so I'd add an ISB there with a comment explaining why it's necessary.